### PR TITLE
Refactory - Wallet Selector Modal + unit test

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
Describe your changes and implementation choices. More details make PRs easier to review.

Remove Mui
Reuse Modal Interface
separate Styles and types externally
Adding Unit Test :

Test cases have been loaded to check the state of the element when it is open.
A test case has been added to check the dispatch of type 'UI/MODAL/CLOSE'.
Test cases have been loaded to check that the component does not exist when the state is not open.

### CODE REVIEW

File: src/containers/modals/walletSelector/WalletSelectorModal.tsx [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
+  IconContainer,
+  StyledSvg,
+} from './styles'
+
+declare global {
+  interface Window {
+    ethereum?: any
   }
-`
+}
```
Comment (sk-enya) [inline src/containers/modals/walletSelector/WalletSelectorModal.tsx:34]: why do we need to declare global? if needed this is not the right place to do better to add in root level `types.d ` file. or please look into the correct place to add.

File: src/containers/modals/walletSelector/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import React from 'react'
```
Comment (sk-enya) [inline src/containers/modals/walletSelector/index.test.tsx:1]: Along with common test please add the test cases for the click or hover event if any. 

Like in this case we have `handleSetWallet` or `handleWalletClick` please make sure to add test for those functions as well.

**Review by sk-enya — CHANGES_REQUESTED** [review]

File: src/containers/modals/walletSelector/WalletSelectorModal.tsx (hunk 1 above)
Comment (alvaro-ricotta) [reply src/containers/modals/walletSelector/WalletSelectorModal.tsx]: I has been added Global interface based on @metamask/Providers

**Comment by sk-enya** [comment]
@alvaro-ricotta let's get this close, please do the needed changes

**Comment by sk-enya** [comment]
@alvaro-ricotta this PR is way lot behind the main branch let's recreate this PR once again from fresh instead of spending time unecessorly to resolve  conflict.

**Pull request closed without merging**